ledger/simulation/trace.go

```diff
@@ -136,11 +153,6 @@ func makeSimulationResultWithVersion(lastRound basics.Round, request Request, ve
 	}, nil
 }
 
-func makeSimulationResult(lastRound basics.Round, request Request) Result {
-	result, err := makeSimulationResultWithVersion(lastRound, request, ResultLatestVersion)
-	if err != nil {
-		// this should never happen, since we pass in ResultLatestVersion
-		panic(err)
-	}
-	return result
+func makeSimulationResult(lastRound basics.Round, request Request, nodeConfig config.Local) (Result, error) {
+	return makeSimulationResultWithVersion(lastRound, request, nodeConfig, ResultLatestVersion)
 }
```

Comment:
This makes me wonder if we should merge the 2 functions into one.

Reply:
That's probably a good idea. `makeSimulationResultWithVersion` is never used anywhere else, and I think it's reasonable to only ever support creating a `Result` with the latest version